Fix notice style for not logged in users in learning mode

Imran92 opened the pull request:
Resolves https://github.com/Automattic/sensei/issues/7242

## Proposed Changes
* Updated styles of the notice that's shown to the user on the lesson page in learning mode when the user is not logged in.

## Testing Instructions
<!--
Add as many details as possible to help others reproduce the issue and test the changes.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

We don't have an exact design for this notice, so we followed the other notices we have that has two buttons.

1. Create a Course with a lesson, learning mode should be enabled for the lesson
3. Try to access the lesson without logging in
4. Make sure the notice looks as expected
5. Check with all variations of course theme
6. Check with other themes
7. Check in mobile view

#### Desktop
![Screenshot 2023-11-03 at 10 43 04 PM](https://github.com/Automattic/sensei/assets/6820724/799bce84-14a1-40b5-b03c-b3ce9df8f02a)

#### Mobile
![Screenshot 2023-11-03 at 10 43 41 PM](https://github.com/Automattic/sensei/assets/6820724/847c33c8-bbac-46ef-9de1-b7128a43c938)

## Pre-Merge Checklist
<!-- Complete applicable items on this checklist **before** merging. Items that are not applicable can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed. -->
- [x] PR title and description contain sufficient detail and accurately describe the changes
- [x] Acceptance criteria is met
- [ ] Decisions are publicly documented
- [x] Adheres to coding standards ([PHP](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/), [JavaScript](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/), [CSS](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/css/), [HTML](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/html/))
- [ ] All strings are translatable (without concatenation, handles plurals)
- [ ] Follows our naming conventions (P6rkRX-4oA-p2)
- [ ] Hooks (p6rkRX-1uS-p2) and functions are documented
- [x] New UIs are responsive and use a [mobile-first approach](https://zellwk.com/blog/how-to-write-mobile-first-css/)
- [ ] New UIs match the designs
- [ ] Different user privileges (admin, teacher, subscriber) are tested as appropriate
- [ ] Code is tested on the minimum supported PHP and WordPress versions
- [ ] User interface changes have been tested on the latest versions of Chrome, Firefox and Safari
- [ ] "Needs Documentation" label is added if this change requires updates to documentation
- [ ] Known issues are created as new GitHub issues

Imran92 committed:
Add additional classes to the notice buttons

Imran92 committed:
Add font size for default variation

Imran92 committed:
Add styles for the actions in the notice

Imran92 committed:
Add changelog

Imran92 committed:
Merge branch 'fix/not-enrolled-notice-style-in-lm' into fix/not-logged-in-style-in-lm

Imran92 committed:
Fix font size of sign in button

Imran92 committed:
Move style to variation specific file

merkushin commented:
Looks better and more consistent now, but styles are a bit different. I took the example from the issue with a quiz.
In the quiz notice the header is larger (font-size).
![CleanShot 2023-11-07 at 18 34 52@2x](https://github.com/Automattic/sensei/assets/329356/d614c374-d75c-4b2d-a31c-627c159aa39a)
![CleanShot 2023-11-07 at 18 34 44@2x](https://github.com/Automattic/sensei/assets/329356/475163bf-08c3-4dbe-8912-2cb21085c0d0)

Imran92 commented:
Hi Dmitry 👋 Thanks for taking a look

> Looks better and more consistent now, but styles are a bit different. I took the example from the issue with a quiz.
> In the quiz notice the header is larger (font-size).

Right, actually we don't have an exact design for this specific notice. For example, we don't have a small Title, like "Awaiting Grade", rather it's a long text. So making it h2 sized makes the font look a bit too big. So I didn't change the font size intentionally, as we don't have a style to follow exactly. This is how it looks in h2 size. If you think it looks alright, we can change it. LMKWYT
![image](https://github.com/Automattic/sensei/assets/6820724/fc9b65bd-d1d3-4530-857f-73bcca5f5dbd)
![image](https://github.com/Automattic/sensei/assets/6820724/323765a5-27ca-421a-ac41-e38167921b89)
![image](https://github.com/Automattic/sensei/assets/6820724/e88f0d60-a66f-4c17-83d5-3fd185019d78)
![image](https://github.com/Automattic/sensei/assets/6820724/8e41dfef-4016-491b-a293-3d2aec6447c3)

Imran92 committed:
Fix background and color Purchase Course button

Imran92 committed:
Use different selector to target link style

merkushin reviewed (dismissed):
Looks good 👍

Imran92 committed:
Merge branch 'trunk' into fix/not-logged-in-style-in-lm

Imran92 committed:
Merge pull request #7285 from Automattic/fix/notice-style-for-paid-courses

Fix notice style for paid courses

donnapep commented:
I'm not sure why (as I don't have any template customizations and I don't appear to have set any global styles either), but this is what I'm seeing for this PR:

![Screenshot 2023-11-15 at 2 38 18 PM](https://github.com/Automattic/sensei/assets/1190420/dd664f9a-11fc-4437-a840-a22e36a067b2)

And when I'm logged out and on the trunk branch to test #7263, I see the following. The button text is the same color as the background and the "Please register" text is too big (because it's an `h2`, more below):

![Screenshot 2023-11-15 at 2 43 58 PM](https://github.com/Automattic/sensei/assets/1190420/42681d30-e6be-4bc1-b39b-c6030a340ac7)

Could I have missed some Course theme changes or anything like that? 🤔 

It's coming from this CSS, which is changing the button text color:

![Screenshot 2023-11-15 at 3 06 37 PM](https://github.com/Automattic/sensei/assets/1190420/a4189a1c-7893-4468-ad9c-0d187102e5d2)

At some point, we seem to have changed the "Please register..." text to an `h2`. I'm guessing this was for styling purposes? The problem is that this is not [semantic HTML](https://www.semrush.com/blog/semantic-html5-guide/) as this text is not a heading. Can we change it back to a `<p>` and fix any styling with CSS? Incidentally, I think it looks just fine when I switch it to a `<p>`:

![Screenshot 2023-11-15 at 3 03 02 PM](https://github.com/Automattic/sensei/assets/1190420/ee49b240-4c1d-40f1-8201-ee36dc904689)

Imran92 committed:
Merge branch 'trunk' into fix/not-logged-in-style-in-lm

Imran92 committed:
Fix button text color

Imran92 committed:
Merge branch 'fix/not-logged-in-style-in-lm' of https://github.com/Automattic/sensei into fix/not-logged-in-style-in-lm

Imran92 committed:
Change the notice text element to paragraph tag

Imran92 committed:
Merge branch 'trunk' into fix/not-logged-in-style-in-lm

Imran92 commented:
Hi @donnapep 👋 Thanks for taking a look!

> I'm not sure why (as I don't have any template customizations and I don't appear to have set any global styles either), but this is what I'm seeing for this PR:

I've fixed it here https://github.com/Automattic/sensei/pull/7264/commits/55ab34fd0f5b2d804b7b99a062794f2d0542521c!

> Can we change it back to a <p> and fix any styling with CSS? Incidentally, I think it looks just fine when I switch it to a <p>:

Thanks for identifying the semantics issue, I've updated it here https://github.com/Automattic/sensei/pull/7264/commits/623672626cff57a69c10b06fe7ceff19654a98f0. The style looks go to me too, so I haven't changed anything else.

donnapep reviewed (approved):
👍🏻
